cfg/builder/builder_walk.cc

```diff
@@ -682,31 +682,32 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::ExpressionPtr &what, BasicBlo
                         auto newCctx = cctx.withTarget(blockrv)
                                            .withBlockBreakTarget(cctx.target)
                                            .withLoopScope(headerBlock, postBlock, true)
-                                           .withSendAndBlockLink(link);
+                                           .withSendAndBlockLink(std::move(link));
                         if (isKernelLambda(s)) {
                             newCctx.isInsideLambda = true;
                         }
                         blockLast = walk(newCctx, s.block()->body, argBlock);
-                    }
```

Comment:
I dunno, I realize that the new approach is more efficient, but the clear signal that we are scoping `newCctx` to just be walking the body of the block is very nice.

Maybe we should keep the scoping and deliberately copy `link` to move it, and keep the `std::move(link)` at the bottom?

Reply:
That's what I had originally done, but realized that the final move would be conditional on the block not terminating in the unreachable block. I'm happy to go back to what we had before, as the additional overhead of the copy isn't worse than what we have now.

I'll revert that change and move in the final use of `link` instead 👍